### PR TITLE
Changelog for 0.34.0

### PR DESCRIPTION
**Pull request by SimonBiggs**
Prep for 0.34.0 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,155 @@ This project adheres to
...
+* Documentation is now stored within the library, moving from `docs` to
+  `lib/pymedphys/docs`
+* Docs can now be built, viewed, and updated with hot-reloading by running
+  `poetry run pymedphys dev docs --live`.
+* The output directory for the docs building can now be controlled by passing
+  an `--output` flag to `pymedphys dev docs`.
+* A new CLI utility that propagates a range of files that depend upon each
+  other files within the repo are "propagated" by calling
+  `poetry run pymedphys dev propagate`. The files created/updated by this
+  command are `lib/pymedphys/_version.py`, the 'extras' field within
```
Comment (SimonBiggs) [inline CHANGELOG.md:145]: Mention the extras field last

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,12 +8,155 @@ This project adheres to
...
+
+
+### Experimental API changes
+
+#### Breaking changes
+
+* Removed `pymedphys experimental gui`.
+
+
+### Developer facing changes
```
Comment (SimonBiggs) [inline CHANGELOG.md:131]:
```suggestion
### Developer facing API changes
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,12 +8,155 @@ This project adheres to
...
+
+* The pymedphys library directory has moved to `lib/pymedphys`.
```
Comment (SimonBiggs) [inline CHANGELOG.md:133]:
```suggestion
* The pymedphys library directory has moved to `lib/pymedphys`.
  After this move you will need to rerun `poetry install -E dev`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,12 +8,156 @@ This project adheres to
...
+  `mudensity` API is still currently available, but it will be removed in a
+  future release.
+* `pymedphys.Delivery.from_logfile` has been renamed as
+  `pymedphys.Delivery.from_trf`. The previous name is still temporarily
+  available but it will be removed in a future release.
+
+#### New features
+
+* Created an online PyMedPhys GUI. It is accessible from
+  https://app.pymedphys.com. This is in its early stages and most parts of the
```
Comment (sjswerdloff) [inline CHANGELOG.md:44]: might want to say the hosted version is intended for demonstration purposes only, and isn't appropriate for "daily use" (better that the facility host locally).
You could put in a query to see if there is interest in a hosted service.

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,12 +8,156 @@ This project adheres to
...
+* The output directory for the docs building can now be controlled by passing
+  an `--output` flag to `pymedphys dev docs`.
+* A new CLI utility that propagates a range of files that depend upon each
+  other files within the repo are "propagated" by calling
+  `poetry run pymedphys dev propagate`. The files created/updated by this
+  command are `lib/pymedphys/_version.py`, the 'extras' field within
+  `pyproject.toml`, `requirements.txt` and `requirements-dev.txt`,
+  `lib/pymedphys/.pylintrc`, `lib/pymedphys/docs/README.rst`, and
+  `lib/pymedphys/docs/CHANGELOG.md`.
+* `pymedphys dev tests` now allow optional simpler flags. `--run-only-slow` can
```
Comment (sjswerdloff) [inline CHANGELOG.md:150]: ...tests` _includes simplified flags_

File: CHANGELOG.md (hunk 4 above)
Comment (SimonBiggs) [reply CHANGELOG.md:44]: Yup, I agree. And I should also add the warning that when using the demo sensitive information should not be uploaded. If the user wants to utilise the tools with sensitive information they should install it on their own machine.

**Comment by SimonBiggs** [comment]
For an insight into what I'm doing here, I'm using this PR facility to be able to step through all the changes since the last release and mark them as "viewed":

image

That way I can write up the changelog in a systematic way.

**Review by sjswerdloff — APPROVED** [review]

**Pull request merged**